### ISSUE TITLE
building from source fails (go1.5, test for runtime-gdb)

### ISSUE THREAD

**Issue by MartinSteffen** [golang/go#12830]
I tried to build go from the sources. I followed the instruction from 

https://golang.org/doc/install/source

**  What version of Go are you using (go version)?

go version go1.4.2 linux/amd64

and 

go1.4

**  What operating system and processor architecture are you using?

fedora 20 (Linux  3.19.8-100.fc20.x86_64 )

Architecture:          x86_64
CPU op-mode(s):        32-bit, 64-bit
Byte Order:            Little Endian
CPU(s):                4
On-line CPU(s) list:   0-3
Thread(s) per core:    2
Core(s) per socket:    2
Socket(s):             1
NUMA node(s):          1
Vendor ID:             GenuineIntel
CPU family:            6
Model:                 42
Model name:            Intel(R) Core(TM) i5-2520M CPU @ 2.50GHz
Stepping:              7
CPU MHz:               2103.320
CPU max MHz:           3200.0000
CPU min MHz:           800.0000
BogoMIPS:              4988.73
Virtualization:        VT-x
L1d cache:             32K
L1i cache:             32K
L2 cache:              256K
L3 cache:              3072K
NUMA node0 CPU(s):     0-3

**  What did you do?

I did the steps as described by the installation-from-source webpage. 

  - clone the go-language repository
  - execute all.bash in subdir src.

   

**  What did you see?

The tests failed which means, the installation failed. The following steps
went seemingly ok

##### Building packages and commands for linux/amd64.

The following section of the installation failed, namely the testing

##### Testing packages.

in particular

--- FAIL: 
	runtime-gdb_test.go:42: gdb version 7.7
	runtime-gdb_test.go:146: print mapvar failed: $1 = map[string]string = {[<error reading variable: Cannot access memory at address 0xb5cc>] = <error reading variable: Cannot access memory at address 0xb5cc>, [<error reading variable: Cannot access memory at address 0x3>] = <error reading variable: Cannot access memory at address 0x3>}
FAIL
FAIL	runtime	41.748s

which resulted in

2015/10/03 14:54:11 Failed: exit status 1

** Go Versions:

   I made the above attempt for

     git checkout master
     git checkout go1.5

both gave the same error

I also tried go1.4.2 (which is shipped with my fedora version) and I tried
go1.4 (where I downloaded the binared and moved the whole to ~/go1.4). In
all combinations, the result was the same.

However, when I did

      git checkout go1.4

that one went through, i.e., the go1.4 version, I was able to build from the sources.

**Comment by davecheney** [comment]
I wouldn't worry about that gdb failure. If you temporarily remove gdb, the test will be skipped.

**Comment by MartinSteffen** [comment]
yes, thanks, removing gdb skipped the test and the script ran through.  ("disabling" gdb is a radical measure though....). But it did the job.

**Comment by davecheney** [comment]
Yeah, I'm aorry about the workaround. As you probably know gdb and go don't
get along, so your not missing much.

**Comment by MartinSteffen** [comment]
Sure, no problem. I simply made gdb non-executable temporily and worked just fine.

**Comment by ianlancetaylor** [comment]
What does `gdb --version` print on your system?

According to #11214, gdb 7.7 is sufficient on Ubuntu, and that is what the test looks for.  Though it's possible that Ubuntu has patches that Fedora does not.

**Comment by rsc** [comment]
Still awaiting reply from Oct 3. Too late for Go 1.6.
